osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/tasks/EventCleanupJob.java

```diff
@@ -0,0 +1,153 @@
+ * converted to CSV file and stored to the file system. Then the old events are
+ * deleted from the database.
+ */
+@DisallowConcurrentExecution
+public class EventCleanupJob extends CsvWriterJob implements Job {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventCleanupJob.class);
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.retention}")
+    private int eventRetentionPeriodInMonths;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.csv.file.location}")
+    private String csvFileLocation;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.csv.file.prefix}")
+    private String csvFilePrefix;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.csv.file.compression.enabled}")
+    private boolean csvFileCompressionEnabled;
+
+    @Value("${osgp.scheduling.job.database.cleanup.event.page.size}")
+    private int eventPageSize;
+
+    @Autowired
+    private TransactionalEventService transactionalEventService;
+
+    @Override
+    public void execute(final JobExecutionContext context) throws JobExecutionException {
+        LOGGER.info("Quartz triggered cleanup of database - event records.");
+        final DateTime start = DateTime.now();
+
+        final Date retention = this.calculateDate();
+        final List<Event> oldEvents = this.transactionalEventService.getEventsBeforeDate(retention, this.eventPageSize);
+        if (!oldEvents.isEmpty()) {
+            this.saveEventsToCsvFile(oldEvents);
+
+            LOGGER.info("Deleting events...");
+            this.transactionalEventService.deleteEvents(oldEvents);
+
+            System.gc();
+        }
+
+        final DateTime end = DateTime.now();
+        LOGGER.info("Start: {}, end: {}, duration: {} milliseconds.", start, end, end.getMillis() - start.getMillis());
+    }
+
+    private Date calculateDate() {
+        final Date date = DateTime.now().minusMonths(this.eventRetentionPeriodInMonths).toDate();
+        LOGGER.info("Determined date: {} based on event retention period in months: {}.", date,
+                this.eventRetentionPeriodInMonths);
+
+        return date;
+    }
+
+    private void saveEventsToCsvFile(final List<Event> events) {
+        try {
+            LOGGER.info("Converting events...");
+            final String[] header = EventToStringArrayConverter.getEventFieldNames();
+            final List<String[]> lines = EventToStringArrayConverter.convertEvents(events);
+            LOGGER.info("Events converted.");
+
+            final String csvFilePath = this.writeCsvFile(this.csvFileLocation, this.csvFilePrefix, header, lines);
+
+            if (this.csvFileCompressionEnabled) {
+                this.compressCsvFile(csvFilePath);
+            }
+        } catch (final Exception e) {
+            LOGGER.error("Exception during CSV file creation, compression or event deletion.", e);
+        }
+    }
+
+    private final static class EventToStringArrayConverter {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a private constructor to hide the implicit public one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1118)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AModifiersOrderCheck)